### ISSUE TITLE
Trying to REPRODUCIBLY build Passport, running into 'Just' related errors. (WalletScrutiny)

### ISSUE THREAD

**Issue by xrviv** [Foundation-Devices/passport2#501]
### The current firmware version on your Passport.

2.2.0

### Passport Model

Passport Batch 2

### Software Wallet Details

Hello, I'm Danny from WalletScrutiny, 

I've been trying to follow the [instructions](https://github.com/Foundation-Devices/passport2/blob/main/REPRODUCIBILITY.md) for reproducing Passport2 v2.2.0 (I've also tried 2.3.0 - same issues.) 

Just as I run `$ just build-docker`

I get:

```
danny@lw10:~/work/builds/hardware/passport2$ just build-docker
docker build -t ${DOCKER_IMAGE} .
Sending build context to Docker daemon  613.6MB
Step 1/12 : FROM ubuntu:20.04
 ---> 2abc4dfd8318
Step 2/12 : ARG DEBIAN_FRONTEND=noninteractive
 ---> Using cache
 ---> 03a3b1b5a63c
Step 3/12 : RUN apt-get update &&     apt-get install -y automake                        autotools-dev                        build-essential                        curl                        gcc-arm-none-eabi                        git                        libffi-dev                        libssl-dev                        libtool                        pkg-config                        pycodestyle                        python3                        python3-pip &&     rm -rf /var/lib/apt/lists/*
 ---> Using cache
 ---> dc534010a0a6
Step 4/12 : RUN pip3 install reuse
 ---> Using cache
 ---> a3fec50fb507
Step 5/12 : ENV RUSTUP_HOME="/rustup"
 ---> Using cache
 ---> 0db3622d0c31
Step 6/12 : ENV CARGO_HOME="/cargo"
 ---> Using cache
 ---> a8410d4d058a
Step 7/12 : RUN mkdir -p /rustup /cargo &&     curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs |         sh -s -- -y --profile minimal --default-toolchain 1.67.1
 ---> Using cache
 ---> 47645745ad68
Step 8/12 : ENV PATH="/cargo/bin:${PATH}"
 ---> Using cache
 ---> c8a59a0fe6e1
Step 9/12 : RUN rustup component add clippy &&     rustup component add rustfmt &&     rustup target add thumbv7em-none-eabihf
 ---> Using cache
 ---> 7a02ab61f9e9
Step 10/12 : RUN cargo install cbindgen --version "^0.24"
 ---> Using cache
 ---> 7be40032dfcc
Step 11/12 : RUN cargo install just --version "^1.26" &&     mv /cargo/bin/cbindgen /usr/local/bin/cbindgen &&     mv /cargo/bin/just /usr/local/bin/just &&     chmod 755 /usr/local/bin/cbindgen &&     chmod 755 /usr/local/bin/just
 ---> Running in 4e5f8c6f2748
    Updating crates.io index
 Downloading crates ...
  Downloaded just v1.26.0
  Installing just v1.26.0
error: failed to compile `just v1.26.0`, intermediate artifacts can be found at `/tmp/cargo-installUn9hbg`

Caused by:
  failed to select a version for `env_logger`.
      ... required by package `just v1.26.0`
  versions that meet the requirements `^0.11.0` are: 0.11.3, 0.11.2, 0.11.1, 0.11.0

  the package `just` depends on `env_logger`, with features: `anstream` but `env_logger` does not have these features.
   It has an optional dependency with that name, but that dependency uses the "dep:" syntax in the features table, so it does not have an implicit feature with that name.


  failed to select a version for `env_logger` which could resolve this conflict
The command '/bin/sh -c cargo install just --version "^1.26" &&     mv /cargo/bin/cbindgen /usr/local/bin/cbindgen &&     mv /cargo/bin/just /usr/local/bin/just &&     chmod 755 /usr/local/bin/cbindgen &&     chmod 755 /usr/local/bin/just' returned a non-zero code: 101
error: Recipe `build-docker` failed on line 10 with exit code 101
```
This is run in our build server. I would appreciate some assistance, thank you.

### Expected Behavior

Should build the firmware.

### Please provide the steps that you've used to reproduce the issue

1. ssh into our debian server
2. `git clone https://github.com/Foundation-Devices/passport2.git`
3. `cd passport2`
4. `git checkout v2.1.2`
5. Install Docker
6. Install Podman
7. Install Just using these instructions:

```
# create ~/bin
mkdir -p ~/bin

# download and extract just to ~/bin/just
curl --proto '=https' --tlsv1.2 -sSf https://just.systems/install.sh | bash -s -- --to ~/bin

# add `~/bin` to the paths that your shell searches for executables
# this line should be added to your shells initialization file,
# e.g. `~/.bashrc` or `~/.zshrc`
export PATH="$PATH:$HOME/bin"

# just should now be executable
just --help
```
8. `$ just clean`
10. `$ just build-docker`

### Impact of the issue

{"label"=>"Low: Minor issue with minimal impact on usability."}

### If you found any similar or related issues, provide links to them here.

None

**Comment by jeandudey** [comment]
Hello @xrviv,

To reproduce 2.1.2 you'd need to add the `--locked` argument to `cargo install ...` commands in the `Dockerfile`, so you'll need to edit it before creating the image, this is because some dependencies of just require a newer compiler and at the time it worked but Cargo chooses to use newer versions of dependencies when available so that's why it is breaking, `--locked` changes that behavior.

On the [v2.3.0-reproducibility](https://github.com/Foundation-Devices/passport2/tree/v2.3.0-reproducibility) tag we have fixed this for `v2.3.0` (and any future versions).

@sethforprivacy has made a MR to update the build script of Passport for WalletScrutiny:

https://gitlab.com/walletscrutiny/walletScrutinyCom/-/merge_requests/654

See also:

https://github.com/Foundation-Devices/passport2/issues/493#issuecomment-2059487820

Feel free to re-open the issue if there's any other error when reproducing `v2.3.0`.

**Comment by xrviv** [comment]
Thank you. Will try this now.
